### PR TITLE
feat: implement low-level drawLine

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**

### CODE REVIEW

File: src/draw/drawLine.ts [hunk 1]
```diff
@@ -26,6 +26,10 @@ export interface DrawLineOptions {
    * Image to which the resulting image has to be put.
    */
   out?: IJS;
+  /**
+   * Enable low-level drawing.
+   */
+  low?: boolean;
```
Comment (targos) [inline src/draw/drawLine.ts:32]: I'm not sure about this. The `low` option name isn't very self-explanatory and there is now a contradiction (if you pass `low: true` and `out`, ``out is ignored). 
@stropitek What do you think? This is to resolve https://github.com/image-js/image-js-typescript/issues/90

Comment (wadjih-bencheikh18) [reply src/draw/drawLine.ts:32]: Maybe i can use type
`{Low:boolean}||{other params}`

Comment (stropitek) [reply src/draw/drawLine.ts:32]: I don't know exactly how `getOutputImage` works.

I also don't understand what "low" means in this context. Would call it `clone` with a default value of false. Don't we have the option to create a new image in other methods as well @targos? API and default value should be consistent it that case

Comment (stropitek) [reply src/draw/drawLine.ts:32]: Would say we can remove the `out` option. If image is cloned `out` image is cloned by the function.

Comment (targos) [reply src/draw/drawLine.ts:32]: When I opened https://github.com/image-js/image-js-typescript/issues/90, I was thinking of making a private low-level API which would never take care or have the option to clone the original image. To me this is just about optimizing for the next methods that will draw a lot of lines with the same color/other options.

Comment (stropitek) [reply src/draw/drawLine.ts:32]: ok got it, SGTM

maybe later we could expose those low-level method in a callback:
```js
const newImage = image.annotate((img) => {
  img.drawLine();
  img.drawCircle();
});
```

Comment (wadjih-bencheikh18) [reply src/draw/drawLine.ts:32]: maybe we can implement a low-level drawline function(without any option params) and use it in original drawline

**Comment by wadjih-bencheikh18** [comment]
@targos could i work on this ?
also would you please specifie how do you want me to impliment it ( I like daniel suggestion )

**Comment by targos** [comment]
I think we need to design the API clearly before working on the code. I'll prepare a hackmd where we can do that collaboratively.

**Pull request closed without merging**